libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -521,7 +522,7 @@ void AP_Baro::update(void)
             // update altitude calculation
             float ground_pressure = sensors[i].ground_pressure;
             if (!is_positive(ground_pressure) || isnan(ground_pressure) || isinf(ground_pressure)) {
-                sensors[i].ground_pressure = sensors[i].pressure;
```

Comment:
> this would create a baro pressure reading that throws the aircraft up possibly thousands of feet.

With the current code it can throw the aircraft down possibly thousands of feet...

> Ground pressure is just the relative starting point for all altitude caluclations, you simply can't mess with it in flight ever

The current code does...

@WickedShell @magicrub I'm not sure you have seen this code correctly. The only thing this change does is that if the current ground pressure is a bad value it will replace it with the MSL pressure instead of the pressure the sensor has at the time.
In my view, none of the two approaches are good - the issue comes from the fact that ground pressure is a parameter and so a user can set it to a bad value during flight. In my view, we should instead save the ground pressure value every time it is good and re-use it here if it is bad. In complement besides the `sensors[i].healthy` check above, we should check the sensor was calibrated.